### ISSUE TITLE
[RLlib] pip install ray[rllib]==2.0.1 fails due to missing [rllib] extra

### ISSUE THREAD

**Issue by ChrisMcCarthyDev** [ray-project/ray#29652]
### What happened + What you expected to happen

I have a Python package that has `install_requires=['ray[rllib]']` as part of the setup.py file. My CI/CD pipeline (Running on Linux VM in AWS.) has broken down since the release of ray 2.0.1 as I'm now getting this error message:
 ```
pkg_resources.UnknownExtra: ray 2.0.1 has no such extra feature 'rllib'
```

When replicating in a Windows environment, I received the error message:
```
ERROR: Could not find a version that satisfies the requirement ray[rllib] (from versions: none)
ERROR: No matching distribution found for ray[rllib]
```

### Linux Error
Ray = ray 2.0.1 with [rllib] extra
Python = 3.8 (Only major and minor version known)
OS = Linux, version 5.15.0-1020-aws

### Windows Error
Ray = ray 2.0.1 with [rllib] extra
Python = 3.10.7
OS = Windows 10 Pro (10.0.19044 Build 19044)

### Linux Environment

```
mkdir ray_201_test
cd ray_201_test
python3 -m venv venv
venv/bin/pip install ray[rllib]
```

### Windows Environment
```
mkdir ray_201_test
cd .\ray_201_test\
python3 -m venv venv
.\venv\Scripts\pip install ray[rllib]
```

### Issue Severity

Low: It annoys or frustrates me.

**Comment by avnishn** [comment]
I don't have access to a windows machine, however, I attempted to repro this issue on MacOS M1, a linux instance via google colab, and a linux instance on gitpod.io. 

I wasn't able to reproduce your issue running your script. 

Are you still running into this issue?

**Comment by ChrisMcCarthyDev** [comment]
So I've done some more investigating. It looks as if the issue is only with Python 3.10, and is ray as a whole, not just ray with [rllib] extra. 

### Windows 10 with Python 3.10.7
```
Microsoft Windows [Version 10.0.19044.2130]
(c) Microsoft Corporation. All rights reserved.

C:\Users\ChristopherMcCarthy>echo  %date%-%time%
 26/10/2022- 8:20:28.84

C:\Users\ChristopherMcCarthy>mkdir ray_201_test

C:\Users\ChristopherMcCarthy>cd .\ray_201_test\

C:\Users\ChristopherMcCarthy\ray_201_test>python3 -m venv venv

C:\Users\ChristopherMcCarthy\ray_201_test>.\venv\Scripts\pip install ray[rllib]
ERROR: Could not find a version that satisfies the requirement ray[rllib] (from versions: none)
ERROR: No matching distribution found for ray[rllib]

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: C:\Users\ChristopherMcCarthy\ray_201_test\venv\Scripts\python3.exe -m pip install --upgrade pip
C:\Users\ChristopherMcCarthy\ray_201_test>C:\Users\ChristopherMcCarthy\ray_201_test\venv\Scripts\python3.exe -m pip install --upgrade pip
Requirement already satisfied: pip in c:\users\christophermccarthy\ray_201_test\venv\lib\site-packages (22.2.2)
Collecting pip
  Using cached pip-22.3-py3-none-any.whl (2.1 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 22.2.2
    Uninstalling pip-22.2.2:
      Successfully uninstalled pip-22.2.2
Successfully installed pip-22.3

C:\Users\ChristopherMcCarthy\ray_201_test>.\venv\Scripts\pip install ray[rllib]
ERROR: Could not find a version that satisfies the requirement ray[rllib] (from versions: none)
ERROR: No matching distribution found for ray[rllib]

C:\Users\ChristopherMcCarthy\ray_201_test>.\venv\Scripts\pip install ray
ERROR: Could not find a version that satisfies the requirement ray (from versions: none)
ERROR: No matching distribution found for ray

C:\Users\ChristopherMcCarthy\ray_201_test>.\venv\Scripts\python
Python 3.10.7 (tags/v3.10.7:6cc6b13, Sep  5 2022, 14:08:36) [MSC v.1933 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> exit()
```

### Windows 10 with Python 3.9.13
```
Microsoft Windows [Version 10.0.19044.2130]
(c) Microsoft Corporation. All rights reserved.

C:\Users\ChristopherMcCarthy>echo  %date%-%time%
 26/10/2022- 8:25:43.88

C:\Users\ChristopherMcCarthy>pyenv install 3.9.13
:: [Info] ::  Mirror: https://www.python.org/ftp/python
:: [Downloading] ::  3.9.13 ...
:: [Downloading] ::  From https://www.python.org/ftp/python/3.9.13/python-3.9.13-amd64.exe
:: [Downloading] ::  To   C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\install_cache\python-3.9.13-amd64.exe
:: [Installing] ::  3.9.13 ...
:: [Info] :: completed! 3.9.13

C:\Users\ChristopherMcCarthy>mkdir ray_201_py_3_9_test

C:\Users\ChristopherMcCarthy>cd ray_201_py_3_9_test

C:\Users\ChristopherMcCarthy\ray_201_py_3_9_test>..\.pyenv\pyenv-win\versions\3.9.13\Scripts\pip install ray[rllib]
Collecting ray[rllib]
  Downloading ray-2.0.1-cp39-cp39-win_amd64.whl (20.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 20.7/20.7 MB 15.2 MB/s eta 0:00:00
Collecting attrs
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting msgpack<2.0.0,>=1.0.0
  Downloading msgpack-1.0.4-cp39-cp39-win_amd64.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.5/62.5 KB ? eta 0:00:00
Collecting numpy>=1.19.3
  Downloading numpy-1.23.4-cp39-cp39-win_amd64.whl (14.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.7/14.7 MB 14.6 MB/s eta 0:00:00
Collecting pyyaml
  Downloading PyYAML-6.0-cp39-cp39-win_amd64.whl (151 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 151.6/151.6 KB 9.4 MB/s eta 0:00:00
Collecting jsonschema
  Using cached jsonschema-4.16.0-py3-none-any.whl (83 kB)
Collecting aiosignal
  Using cached aiosignal-1.2.0-py3-none-any.whl (8.2 kB)
Collecting grpcio<=1.43.0,>=1.32.0
  Downloading grpcio-1.43.0-cp39-cp39-win_amd64.whl (3.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.4/3.4 MB 15.3 MB/s eta 0:00:00
Collecting virtualenv
  Downloading virtualenv-20.16.6-py3-none-any.whl (8.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.8/8.8 MB 16.5 MB/s eta 0:00:00
Collecting filelock
  Using cached filelock-3.8.0-py3-none-any.whl (10 kB)
Collecting requests
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting frozenlist
  Downloading frozenlist-1.3.1-cp39-cp39-win_amd64.whl (34 kB)
Collecting protobuf<4.0.0,>=3.15.3
  Downloading protobuf-3.20.3-cp39-cp39-win_amd64.whl (904 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 904.2/904.2 KB 19.0 MB/s eta 0:00:00
Collecting click<=8.0.4,>=7.0
  Using cached click-8.0.4-py3-none-any.whl (97 kB)
Collecting dm-tree
  Downloading dm_tree-0.1.7-cp39-cp39-win_amd64.whl (90 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.3/90.3 KB ? eta 0:00:00
Collecting tensorboardX>=1.9
  Using cached tensorboardX-2.5.1-py2.py3-none-any.whl (125 kB)
Collecting scikit-image
  Downloading scikit_image-0.19.3-cp39-cp39-win_amd64.whl (12.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.1/12.1 MB 21.1 MB/s eta 0:00:00
Collecting scipy
  Downloading scipy-1.9.3-cp39-cp39-win_amd64.whl (40.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.2/40.2 MB 15.3 MB/s eta 0:00:00
Collecting tabulate
  Using cached tabulate-0.9.0-py3-none-any.whl (35 kB)
Collecting lz4
  Downloading lz4-4.0.2-cp39-cp39-win_amd64.whl (98 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.9/98.9 KB ? eta 0:00:00
Collecting gym<0.24.0,>=0.21.0
  Using cached gym-0.23.1.tar.gz (626 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting matplotlib!=3.4.3
  Downloading matplotlib-3.6.1-cp39-cp39-win_amd64.whl (7.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.2/7.2 MB 12.1 MB/s eta 0:00:00
Collecting pandas
  Downloading pandas-1.5.1-cp39-cp39-win_amd64.whl (10.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.9/10.9 MB 15.6 MB/s eta 0:00:00
Collecting colorama
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting six>=1.5.2
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting importlib-metadata>=4.10.0
  Downloading importlib_metadata-5.0.0-py3-none-any.whl (21 kB)
Collecting cloudpickle>=1.2.0
  Using cached cloudpickle-2.2.0-py3-none-any.whl (25 kB)
Collecting gym-notices>=0.0.4
  Using cached gym_notices-0.0.8-py3-none-any.whl (3.0 kB)
Collecting contourpy>=1.0.1
  Downloading contourpy-1.0.5-cp39-cp39-win_amd64.whl (161 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 162.0/162.0 KB 10.1 MB/s eta 0:00:00
Collecting pillow>=6.2.0
  Downloading Pillow-9.2.0-cp39-cp39-win_amd64.whl (3.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 11.6 MB/s eta 0:00:00
Collecting kiwisolver>=1.0.1
  Downloading kiwisolver-1.4.4-cp39-cp39-win_amd64.whl (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.4/55.4 KB ? eta 0:00:00
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting fonttools>=4.22.0
  Using cached fonttools-4.38.0-py3-none-any.whl (965 kB)
Collecting packaging>=20.0
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting pyparsing>=2.2.1
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting protobuf<4.0.0,>=3.15.3
  Downloading protobuf-3.20.1-cp39-cp39-win_amd64.whl (904 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 904.1/904.1 KB 19.4 MB/s eta 0:00:00
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Downloading pyrsistent-0.18.1-cp39-cp39-win_amd64.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.6/61.6 KB 3.4 MB/s eta 0:00:00
Collecting pytz>=2020.1
  Using cached pytz-2022.5-py2.py3-none-any.whl (500 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting imageio>=2.4.1
  Using cached imageio-2.22.2-py3-none-any.whl (3.4 MB)
Collecting networkx>=2.2
  Using cached networkx-2.8.7-py3-none-any.whl (2.0 MB)
Collecting tifffile>=2019.7.26
  Using cached tifffile-2022.10.10-py3-none-any.whl (210 kB)
Collecting PyWavelets>=1.1.1
  Downloading PyWavelets-1.4.1-cp39-cp39-win_amd64.whl (4.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.2/4.2 MB 14.0 MB/s eta 0:00:00
Collecting distlib<1,>=0.3.6
  Using cached distlib-0.3.6-py2.py3-none-any.whl (468 kB)
Collecting platformdirs<3,>=2.4
  Using cached platformdirs-2.5.2-py3-none-any.whl (14 kB)
Collecting zipp>=0.5
  Using cached zipp-3.10.0-py3-none-any.whl (6.2 kB)
Building wheels for collected packages: gym
  Building wheel for gym (pyproject.toml) ... done
  Created wheel for gym: filename=gym-0.23.1-py3-none-any.whl size=701358 sha256=ddca57c0a66000c0036da03d1f3b929733c2ebdd95f5c9e689ef2eb75055ab05
  Stored in directory: c:\users\christophermccarthy\appdata\local\pip\cache\wheels\4e\be\7e\92a54668db96883e38ce60a9249dc55de7cd6eee49e7311940
Successfully built gym
Installing collected packages: pytz, msgpack, gym-notices, dm-tree, distlib, zipp, urllib3, tabulate, six, pyyaml, pyrsistent, pyparsing, protobuf, platformdirs, pillow, numpy, networkx, lz4, kiwisolver, idna, frozenlist, fonttools, filelock, cycler, colorama, cloudpickle, charset-normalizer, certifi, attrs, virtualenv, tifffile, tensorboardX, scipy, requests, PyWavelets, python-dateutil, packaging, jsonschema, importlib-metadata, imageio, grpcio, contourpy, click, aiosignal, scikit-image, ray, pandas, matplotlib, gym
  WARNING: The script tabulate.exe is installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script f2py.exe is installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts fonttools.exe, pyftmerge.exe, pyftsubset.exe and ttx.exe are installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script normalizer.exe is installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script virtualenv.exe is installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts lsm2bin.exe, tiff2fsspec.exe, tiffcomment.exe and tifffile.exe are installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script jsonschema.exe is installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts imageio_download_bin.exe and imageio_remove_bin.exe are installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script skivi.exe is installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts ray-operator.exe, ray.exe, rllib.exe, serve.exe and tune.exe are installed in 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed PyWavelets-1.4.1 aiosignal-1.2.0 attrs-22.1.0 certifi-2022.9.24 charset-normalizer-2.1.1 click-8.0.4 cloudpickle-2.2.0 colorama-0.4.6 contourpy-1.0.5 cycler-0.11.0 distlib-0.3.6 dm-tree-0.1.7 filelock-3.8.0 fonttools-4.38.0 frozenlist-1.3.1 grpcio-1.43.0 gym-0.23.1 gym-notices-0.0.8 idna-3.4 imageio-2.22.2 importlib-metadata-5.0.0 jsonschema-4.16.0 kiwisolver-1.4.4 lz4-4.0.2 matplotlib-3.6.1 msgpack-1.0.4 networkx-2.8.7 numpy-1.23.4 packaging-21.3 pandas-1.5.1 pillow-9.2.0 platformdirs-2.5.2 protobuf-3.20.1 pyparsing-3.0.9 pyrsistent-0.18.1 python-dateutil-2.8.2 pytz-2022.5 pyyaml-6.0 ray-2.0.1 requests-2.28.1 scikit-image-0.19.3 scipy-1.9.3 six-1.16.0 tabulate-0.9.0 tensorboardX-2.5.1 tifffile-2022.10.10 urllib3-1.26.12 virtualenv-20.16.6 zipp-3.10.0
WARNING: You are using pip version 22.0.4; however, version 22.3 is available.
You should consider upgrading via the 'C:\Users\ChristopherMcCarthy\.pyenv\pyenv-win\versions\3.9.13\python.exe -m pip install --upgrade pip' command.

C:\Users\ChristopherMcCarthy\ray_201_py_3_9_test>
```

**Comment by avnishn** [comment]
@ChrisMcCarthyDev I'm going to close this for now, but please let me know if there are still issues on the current ray versions.
